### ISSUE TITLE
Add support for staged payments to StakedExpenditures

### ISSUE THREAD

**Issue by kronosapiens** [JoinColony/colonyNetwork#1129]
Add ability to create a "staged payment", where the owner can "release" payments by setting claim delays to zero (using the arbitration permission)

**Comment by kronosapiens** [comment]
@arrenv looking at the spec here: https://www.figma.com/file/5V8pr7iMwXsT9L3VAZsmUt/Colony-Dashboard?node-id=834-183859&t=2XO3X0wG35n8bAL5-0

I am curious whether the user is able to make arbitrary edits to the expenditure before locking, or whether the intention is that the user submits all the stages at the time of creation.

Another way, if we give ownership over to the user, they could potentially have some stages set to different recipients. Is this ok?

My preference would be for the user to submit all stages at the time of creation, or alternatively to provide a separate "add stage" function that the user calls through the extension. I would not want to transfer the expenditure to the user for arbitrary editing, as I worry this would allow the user to break the staged payments feature.

**Comment by arrenv** [comment]
> I am curious whether the user is able to make arbitrary edits to the expenditure before locking, or whether the intention is that the user submits all the stages at the time of creation.

Yes, the user is still able to make arbitrary edits before locking. As we get this feature dev ready, the UI will better distinguish locked fields.

> Another way, if we give ownership over to the user, they could potentially have some stages set to different recipients. My understanding is that for staged payments, there should be only one recipient, and the claim delays should not be editable by the user.

Interesting idea, my thoughts are that this could complicate the feature.

> My preference would be for the user to submit all stages at the time of creation, or alternatively to provide a separate "add stage" function that the user calls through the extension before the expenditure is locked. I would not want to transfer the expenditure to the user for arbitrary editing, as I worry this would allow the user to break the staged payments feature.

The idea is certainly that you submit all stages at the time of creation, however by allowing editing before locking, it gives the creator a lower risk chance to get collective input and make corrections before putting their stake at risk in a locked state. With a separate "add stage" function, would a separate "remove stage" also make sense? Would this be instead of an "edit stage" option.

**Comment by kronosapiens** [comment]
Ok, in that case I'll leave it less rigid. User will be able to make arbitrary edits, and release the slots once the expenditure is finalized.

**Comment by area** [comment]
>>    Another way, if we give ownership over to the user, they could potentially have some stages set to different recipients. My understanding is that for staged payments, there should be only one recipient, and the claim delays should not be editable by the user.
>Interesting idea, my thoughts are that this could complicate the feature.

The important point here is that this is possible on the contracts for all expenditures, so the frontend (in my mind, at least) needs to be able to handle that scenario.
